samples/nextjs/src/lib/page-props-factory.test.ts

```diff
@@ -0,0 +1,122 @@
+import { SitecorePagePropsFactory } from './page-props-factory';
+import { LayoutServiceFactory } from './layout-service-factory';
+import {
+  ComponentPropsService,
+  LayoutService,
+  LayoutServiceData,
+  LayoutPersonalizationUtils,
+  EditingDataService,
+  EditingData,
+} from '@sitecore-jss/sitecore-jss-nextjs';
+import { DictionaryServiceFactory } from './dictionary-service-factory';
+import { expect, spy, use } from 'chai';
+import spies from 'chai-spies';
+
+use(spies);
+
+describe('SitecorePagePropsFactory', () => {
+  let factory: SitecorePagePropsFactory;
+  let layoutPersonalizationUtils: LayoutPersonalizationUtils;
+  const context = {
+    preview: false,
+  };
+  let layoutServiceData: LayoutServiceData;
+  const layoutServiceFactory = new LayoutServiceFactory();
+  const editingDataService = new EditingDataService();
+
+  beforeEach(() => {
+    context.preview = false;
+    layoutServiceData = {
+      sitecore: {
+        context: {},
+        route: {
+          name: 'name',
+          placeholders: {},
+        },
+      },
+    };
+
+    const layoutService: LayoutService = {
+      fetchLayoutData: () => Promise.resolve(layoutServiceData),
+    };
+
+    spy.on(layoutServiceFactory, 'create', () => layoutService);
+
+    const editingData: EditingData = {
+      path: '',
+      language: '',
+      layoutData: layoutServiceData,
+      dictionary: {},
+    };
+    spy.on(editingDataService, 'getEditingData', () => Promise.resolve(editingData));
+
+    layoutPersonalizationUtils = new LayoutPersonalizationUtils();
+
+    factory = new SitecorePagePropsFactory(
+      new ComponentPropsService(),
+      new DictionaryServiceFactory(),
```

Comment:
Do not like that we need to create classes, it is possible to mock them? May be add sinon for this purpose?